default.nix

```diff
@@ -477,9 +477,118 @@ in
+        '';
+      };
+
+      startAt = mkOption {
+        type = types.string;
+        default = "hourly";
+        description = "When or how often the backup should run. Must be in the format described in systemd.time 7.";
+      };
+
+      compression = {
+        method = mkOption {
+          type = types.nullOr (types.enum ["none" "lz4" "zstd" "zlib" "lzma"]);
+          default = null;
+          description = "Leaving this unset allows borg to choose. The default for borg 1.1.4 is lz4.";
+        };
+
+        level = mkOption {
+          type = types.nullOr types.int;
+          default = null;
+          description = ''
+            Denotes the level of compression used by borg.
+            Most methods accept levels from 0 to 9 but zstd which accepts values from 1 to 22.
+            If null the decision is left up to borg.
+          '';
+        };
+        
+        auto = mkOption {
+          type = types.bool;
+          default = false;
+          description = "Leaves it to borg to determine whether an individual file should be compressed.";
+        };
+      };
+
+      encryption = {
+        method = mkOption {
+          type = types.enum [
+            "none"
+            "authenticated"
+            "authenticated-blake2"
+            "repokey"
+            "keyfile"
+            "repokey-blake2"
+            "keyfile-blake2"
+          ];
+          default = "none";
+          description = ''
+            The backup can be encrypted by choosing any other value than 'none'.
+            When using encryption the password / passphrase must be provided in passphraseFile.
           '';
-          };
         };
+        
+        passphraseFile = mkOption {
+          type = types.nullOr types.path;
+          default = null;
+        };
+      };
+
+      name = mkOption {
+        type = types.string;
+        default = "{hostname}-{user}-{now}";
+        description = ''
+          The name of the individual backups as used by borg.
+          Certain placeholders will be replaced by borg.
+        '';
+      };
+
+      locations = mkOption {
+        type = types.listOf types.path;
+        default = [cfg.mailDirectory];
+        description = "The locations that are to be backed up by borg.";
+      };
+
+      extraArgumentsForInit = mkOption {
+        type = types.listOf types.string;
+        default = ["--critical"];
+        description = "Additional arguments to add to the borg init command line.";
+      };
+
+      extraArgumentsForCreate = mkOption {
+        type = types.listOf types.string;
+        default = ["--stats"];
+        description = "Additional arguments to add to the borg create command line.";
+      };
+
+      cmdPreexec = mkOption {
+        type = types.nullOr types.string;
+        default = null;
+        description = ''
+          The command to be executed before each backup operation. This is called prior to borg init.
+        '';
```

Comment:
You could add `export BORG_RSH="ssh -i /path/to/private/key"` as an example.

Reply:
Good idea! I have done so and detailed where these commands are called exactly.